package/src/index.ts

```diff
@@ -20,25 +22,188 @@ export function activateBasicCodeIntel(
             sourcegraph.languages.registerHoverProvider(
                 documentSelector(h.fileExts),
                 {
-                    provideHover: (doc, pos) => h.hover(doc, pos),
+                    provideHover: async (doc, pos) =>
+                        (await hasLSIF(doc))
+                            ? await lsif.provideHover(doc, pos)
+                            : await h.hover(doc, pos),
                 }
             )
         )
         ctx.subscriptions.add(
             sourcegraph.languages.registerDefinitionProvider(
                 documentSelector(h.fileExts),
                 {
-                    provideDefinition: (doc, pos) => h.definition(doc, pos),
+                    provideDefinition: async (doc, pos) =>
+                        (await hasLSIF(doc))
+                            ? await lsif.provideDefinition(doc, pos)
+                            : await h.definition(doc, pos),
                 }
             )
         )
         ctx.subscriptions.add(
             sourcegraph.languages.registerReferenceProvider(
                 documentSelector(h.fileExts),
                 {
-                    provideReferences: (doc, pos) => h.references(doc, pos),
+                    provideReferences: async (doc, pos) =>
+                        (await hasLSIF(doc))
+                            ? await lsif.provideReferences(doc, pos)
+                            : await h.references(doc, pos),
                 }
             )
         )
     }
 }
+
+function repositoryFromDoc(doc: sourcegraph.TextDocument): string {
+    const url = new URL(doc.uri)
+    return url.hostname + url.pathname
+}
+
+function commitFromDoc(doc: sourcegraph.TextDocument): string {
+    const url = new URL(doc.uri)
+    return url.search.slice(1)
+}
+
+function pathFromDoc(doc: sourcegraph.TextDocument): string {
+    const url = new URL(doc.uri)
+    return url.hash.slice(1)
+}
+
+function setPath(doc: sourcegraph.TextDocument, path: string): string {
+    const url = new URL(doc.uri)
+    url.hash = path
+    return url.href
+}
+
+async function send({
```

Comment:
Consider giving this a more explicit name:

```suggestion
async function queryLSIF({
```